### ISSUE TITLE
[V3] Implement new Concept Exercise: concept/classes-intermediate

### ISSUE THREAD

**Issue by Tbhesswebber** [exercism/javascript#943]
This issue describes how to implement the `class` concept exercise for the JavaScript track and is a continuation of exercism/javascript#942 .

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student how to consume classes and their properties/methods both from static and instantiation standpoints within JavaScript. 

## Learning objectives

- Understand the `new` keyword.
- Understand the `static` keyword.
- Understand how to consume instance methods and fields
- Understand how to consume static methods.

## Out of scope

- Factory functions
- Class expressions (declaring a class with a var statement)
- Static properties & static methods
- extends keyword
- Class inheritance with super
- Class properties (`get`, `set`, and `Object.defineProperty`)

## Concepts

- Class instantiation with the `new` keyword
- Accessing class properties from outside of a class
- Calling class methods from outside of a class
- Calling static methods from outside of a class

## Prerequisites

- `classes-intro`
- `booleans`
- `numbers`
- `strings`

### Hints

_Note: most resources also include inheritance and so suggestions are welcome for resources that expressly tackle the goals without venturing out of scope._

- [MDN `static` keyword](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/static)

### After

_Note: most resources also include inheritance and so suggestions are welcome for resources that expressly tackle the goals without venturing out of scope._

- [MDN class fields](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Class_fields) - after because it goes out of scope and touches on TC39 proposals (possibly a bad resource?)

## Representer

No changes needed.

## Analyzer

No changes needed.

## Implementing

See the [`Implementation Guide for JavaScript Concept Exercises`](https://github.com/exercism/v3/blob/master/languages/javascript/reference/implementing-a-concept-exercise.md#how-to-implement-a-rust-concept-exercise) for more details.

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

**Comment by junedev** [comment]
I am closing this issue because we decided to cut concepts/concept exercises differently. See https://github.com/exercism/javascript/issues/951#issuecomment-778834953 for details. Classes are covered here now https://github.com/exercism/javascript/issues/1015 and inheritance will be combined with errors into into one exercise with two concepts.
